src/algorithms/tracking/TrackSeeding.cc

```diff
@@ -157,12 +157,21 @@ std::pair<float, float> eicrecon::TrackSeeding::findRoot(std::tuple<float,float,
 		      * pow(Y0,4)) + square(X0) * Y0 + pow(Y0, 3))
     / (square(X0) + square(Y0));
 
+  //If one of the calculated y values is farther than R from the circle center, return the other pair
+  if( square(R) - square(miny - Y0) < 0 )
+	return std::make_pair(-std::sqrt(square(R) - square(miny2 - Y0)) + X0 , miny2);
+  if( square(R) - square(miny2 - Y0) < 0 )
+	return std::make_pair(std::sqrt(square(R) - square(miny - Y0)) + X0 , miny);
```

Comment:
Why is `square(R) - square(miny2 - Y0)` positive when `square(R) - square(miny - Y0)` is negative? That doesn't seem obvious.

Reply:
I think you are right that it is not guaranteed that one of the sqrts has to have a positive argument. But in that case, both `minx` and `minx2` would have to be on the order of 1e-6, which is unlikely.

In the example event shown above, if I use the values of R, X0, Y0 and then calculate `miny, minx, miny2, and minx2` by hand with higher precision, I get `-0.0054761640, -9114.0365, 1.9580288, 1.1812951e-06`, respectively. The `minx2` value is close to zero, which causes issues with the float precision when subtracting 2 large numbers.

Maybe there is a better way to write this part of the code?

Reply:
In essence
```cpp
  const double R0 = std::hypot(X0, Y0); // ovflw safety
  const double miny_alt = Y0 * (1 + R/R0); // assumes R0 > 0
  const double miny2_alt = Y0 * (1 - R/R0); // assumes R0 > 0
  const double minx_alt = miny_alt * X0/Y0; // assumes Y0 != 0
  const double minx2_alt = miny2_alt * X0/Y0; // assumes Y0 != 0
```
and then pick the one with lowest $r$.

Reply:
Part of me doesn't believe this could really be this simple. But then again, another part of me didn't believe it could be that complicated at the start... So, let me know if I'm overlooking something.

Reply:
Hi Wouter. Thanks for all these suggestions.

I will add a guard for the `R0 = 0 (< epsilon)` case.

Otherwise, we may not need to check both possible points. I think the point

`x = X0 * (1 - R/R0) and y = Y0 * (1- R/R0)`

should always be on the circle and on the line connecting the origin to the circle center. And it will always be closer to the origin than the opposite point

`x = X0 * (1 + R/R0) and y = Y0 * (1+ R/R0)`

Reply:
This should hold whether R0 is larger or smaller than R.

Reply:
I made the changes I suggested. I was able to run 10k muons without any issue. At the seed level, the ACTS loc-a and phi reconstruction get modified only very slightly. The other seed parameters don't change at all, as expected.

<img width="597" alt="image1" src="https://github.com/eic/EICrecon/assets/8701169/a663d2f9-e479-4b60-a1ed-c392c1f90c16">

<img width="614" alt="image2" src="https://github.com/eic/EICrecon/assets/8701169/fb7ee73c-3440-46f3-a648-1a50674bcd27">